git-transport/src/client/blocking_io/ssh.rs

```diff
@@ -56,16 +56,16 @@ pub fn connect(
         _ => return Err(Error::UnsupportedSshCommand(ssh_cmd.into())),
     };
 
-    let host = match user.as_ref() {
-        Some(user) => format!("{}@{}", user, host),
-        None => host.into(),
-    };
-
     let path = git_url::expand_path::for_shell(path);
-    let url = git_url::Url::from_parts(
+    let url_parser = if path.to_string().starts_with('/') {
```

Comment:
Keeping a function pointer is nice! I should remember that.

Reply:
Thanks !
I am rather new at rust but I can definitively feel some Python influence and some patterns like this one feel rather intuitive.